package.json

```diff
@@ -23,24 +23,31 @@
     "node": ">=18.20.4"
   },
   "scripts": {
-    "test": "jest --coverage",
-    "test:watch": "jest --watch",
     "build": "rm -rf ./build && tsc --build src",
+    "test": "npm run test:all",
+    "test:all": "run-s test:format test:types test:lint test:md test:unit test:integration",
+    "test:format": "prettier --check .",
+    "test:types": "tsc --noEmit --project ./src/tsconfig.json",
+    "test:unit": "jest --coverage",
+    "test:unit:watch": "jest --watch",
+    "test:integration": "./tools/integration-test.sh",
+    "test:lint": "eslint -f visualstudio ./src",
+    "test:md": "run-s md:lint md:embedme:verify",
     "commitlint": "commitlint --edit",
     "changelog:build": "conventional-changelog -p angular -i CHANGELOG.md -s",
     "version": "./tools/version.sh",
     "preversion": "./tools/pre-version.sh",
-    "integration-test": "./tools/integration-test.sh",
     "env:md-files": "env MD_FILES=\"\"",
-    "eslint": "eslint -f visualstudio ./src",
-    "md:lint": "markdownlint-cli2 \"./*.md\" \"#CHANGELOG.md\"",
+    "md:lint": "markdownlint-cli2 \"./*.md\" \"./docs/*.md\" \"!CHANGELOG.md\"",
     "md:embedme": "embedme README.md docs/*.md",
+    "md:embedme:verify": "npm run md:embedme -- --verify",
     "md:toc": "markdown-toc -i docs/tutorial.md",
     "md:build-all": "run-s -ln md:toc md:embedme",
-    "md:verify": "npm run md:embedme -- --verify",
-    "ci:all": "npm run pre-push",
-    "pre-push": "run-p -ln test eslint md:verify md:lint",
-    "pre-commit": "pretty-quick --staged && npm run eslint && npm run md:lint",
+    "ci:all": "npm run test:all",
+    "pre-push": "run-p -ln test:unit md:embedme:verify",
+    "pre-commit": "run-s format:staged test:lint md:lint",
+    "format:staged": "npm run format -- --staged",
```

Comment:
not a big deal, but
i propose for "format"-commands don't re-use "foramt" in "format:staged" but copy-paster pretty-quick 2-times instead

```json
    "format:staged": "pretty-quick --staged",
    "format": "pretty-quick",
```
p.s. i decided to write this suggestion, because i have a small pain to quick understand how this two command relates each other :)